Binary files unrar.orig/archive.o and unrar/archive.o differ
Binary files unrar.orig/arcread.o and unrar/arcread.o differ
Binary files unrar.orig/blake2s.o and unrar/blake2s.o differ
Binary files unrar.orig/cmddata.o and unrar/cmddata.o differ
diff -rupN unrar.orig/consio.cpp unrar/consio.cpp
--- unrar.orig/consio.cpp	2017-08-11 15:56:22.000000000 +0200
+++ unrar/consio.cpp	2017-10-15 22:57:22.942800109 +0200
@@ -350,3 +350,70 @@ void OutComment(const wchar *Comment,siz
   }
   mprintf(L"\n");
 }
+
+
+#ifdef __MINT__
+int swprintf(wchar_t *wcs, size_t maxlen, const wchar_t *format, ...)
+{
+	va_list args;
+	va_start(args, format);
+	int res = vswprintf(wcs, maxlen, format, args);
+	va_end(args);
+	return res;
+}
+
+
+int vswprintf(wchar_t *wcs, size_t maxlen, const wchar_t *format, va_list args)
+{
+	char buf[maxlen];
+	int res = vswprintf(wcs, maxlen, format, args);
+	int i;
+	if (res >= 0)
+	{
+		for (i = 0; i < res; i++)
+			wcs[i] = (unsigned char)buf[i];
+		if (res < maxlen)
+			wcs[i] = '\0';
+	}
+	return res;
+}
+
+
+int vfwprintf(FILE *stream, const wchar_t *format, va_list args)
+{
+}
+
+
+long wcstol(const wchar_t *nptr, wchar_t **endptr, int base)
+{
+	const int n = wcslen(nptr);
+	char buf[n + 1];
+	int i;
+	long res;
+	char *end = NULL;
+	for (i = 0; i < n; i++)
+		buf[i] = nptr[i];
+	buf[i] = '\0';
+	res = strtol(buf, &end, base);
+	if (end && endptr)
+		*endptr = (wchar_t *)nptr + (end - buf);
+	return res;
+}
+
+
+wchar_t *fgetws(wchar_t *ws, int n, FILE *stream)
+{
+	char *res;
+	char buf[n];
+	int i;
+	
+	res = fgets(buf, n, stream);
+	if (res == NULL)
+		return NULL;
+	for (i = 0; i < n && buf[i]; i++)
+		ws[i] = (unsigned char)buf[i];
+	ws[i] = '\0';
+	return ws;
+}
+
+#endif
Binary files unrar.orig/consio.o and unrar/consio.o differ
Binary files unrar.orig/crc.o and unrar/crc.o differ
Binary files unrar.orig/crypt.o and unrar/crypt.o differ
Binary files unrar.orig/encname.o and unrar/encname.o differ
Binary files unrar.orig/errhnd.o and unrar/errhnd.o differ
Binary files unrar.orig/extinfo.o and unrar/extinfo.o differ
Binary files unrar.orig/extract.o and unrar/extract.o differ
Binary files unrar.orig/filcreat.o and unrar/filcreat.o differ
Binary files unrar.orig/file.o and unrar/file.o differ
diff -rupN unrar.orig/filefn.cpp unrar/filefn.cpp
--- unrar.orig/filefn.cpp	2017-08-11 15:56:23.000000000 +0200
+++ unrar/filefn.cpp	2017-10-16 01:28:52.213104332 +0200
@@ -154,7 +154,7 @@ int64 GetFreeDisk(const wchar *Name)
       uiUserFree.u.HighPart<=uiTotalFree.u.HighPart)
     return INT32TO64(uiUserFree.u.HighPart,uiUserFree.u.LowPart);
   return 0;
-#elif defined(_UNIX)
+#elif defined(_UNIX) && !defined(__MINT__)
   wchar Root[NM];
   GetFilePath(Name,Root,ASIZE(Root));
   char RootA[NM];
Binary files unrar.orig/filefn.o and unrar/filefn.o differ
Binary files unrar.orig/filestr.o and unrar/filestr.o differ
Binary files unrar.orig/find.o and unrar/find.o differ
Binary files unrar.orig/getbits.o and unrar/getbits.o differ
Binary files unrar.orig/global.o and unrar/global.o differ
Binary files unrar.orig/hash.o and unrar/hash.o differ
Binary files unrar.orig/headers.o and unrar/headers.o differ
Binary files unrar.orig/isnt.o and unrar/isnt.o differ
Binary files unrar.orig/list.o and unrar/list.o differ
Binary files unrar.orig/match.o and unrar/match.o differ
Binary files unrar.orig/options.o and unrar/options.o differ
diff -rupN unrar.orig/os.hpp unrar/os.hpp
--- unrar.orig/os.hpp	2017-10-15 19:31:26.391284984 +0200
+++ unrar/os.hpp	2017-10-15 21:25:50.335392309 +0200
@@ -30,7 +30,7 @@
 #define _WIN32_WINNT 0x0501
 
 #if !defined(ZIPSFX)
-#define RAR_SMP
+#undef RAR_SMP
 #endif
 
 #define WIN32_LEAN_AND_MEAN
@@ -250,4 +250,13 @@
 #define ALLOW_MISALIGNED
 #endif
 
+#ifdef __MINT__
+extern "C" int swprintf(wchar_t *wcs, size_t maxlen, const wchar_t *format, ...);
+extern "C" wchar_t *fgetws(wchar_t *ws, int n, FILE *stream);
+extern "C" int vfwprintf(FILE *stream, const wchar_t *format, va_list args);
+extern "C" int vswprintf(wchar_t *wcs, size_t maxlen, const wchar_t *format, va_list args);
+#define putwchar(c) putchar(c)
+#undef UNIX_TIME_NS
+#endif
+
 #endif // _RAR_OS_
Binary files unrar.orig/pathfn.o and unrar/pathfn.o differ
Binary files unrar.orig/qopen.o and unrar/qopen.o differ
Binary files unrar.orig/rar.o and unrar/rar.o differ
Binary files unrar.orig/rarvm.o and unrar/rarvm.o differ
Binary files unrar.orig/rawread.o and unrar/rawread.o differ
Binary files unrar.orig/rdwrfn.o and unrar/rdwrfn.o differ
Binary files unrar.orig/recvol.o and unrar/recvol.o differ
Binary files unrar.orig/resource.o and unrar/resource.o differ
Binary files unrar.orig/rijndael.o and unrar/rijndael.o differ
Binary files unrar.orig/rs.o and unrar/rs.o differ
Binary files unrar.orig/rs16.o and unrar/rs16.o differ
Binary files unrar.orig/scantree.o and unrar/scantree.o differ
Binary files unrar.orig/secpassword.o and unrar/secpassword.o differ
Binary files unrar.orig/sha1.o and unrar/sha1.o differ
Binary files unrar.orig/sha256.o and unrar/sha256.o differ
Binary files unrar.orig/smallfn.o and unrar/smallfn.o differ
Binary files unrar.orig/strfn.o and unrar/strfn.o differ
Binary files unrar.orig/strlist.o and unrar/strlist.o differ
Binary files unrar.orig/system.o and unrar/system.o differ
Binary files unrar.orig/threadpool.o and unrar/threadpool.o differ
Binary files unrar.orig/timefn.o and unrar/timefn.o differ
Binary files unrar.orig/ui.o and unrar/ui.o differ
Binary files unrar.orig/unicode.o and unrar/unicode.o differ
Binary files unrar.orig/unpack.o and unrar/unpack.o differ
Binary files unrar.orig/unrar and unrar/unrar differ
Binary files unrar.orig/volume.o and unrar/volume.o differ
